src/bootstrap/src/core/build_steps/llvm.rs

```diff
@@ -241,7 +241,7 @@ pub(crate) fn is_ci_llvm_available(config: &Config, asserts: bool) -> bool {
 
 /// Returns true if we're running in CI with modified LLVM (and thus can't download it)
 pub(crate) fn is_ci_llvm_modified(config: &Config) -> bool {
-    CiEnv::is_ci() && config.rust_info.is_managed_git_subrepository() && {
+    CiEnv::is_ci() && config.in_tree_llvm_info.is_managed_git_subrepository() && {
```

Comment:
An alternative solution for the [backtrace CI issue](https://rust-lang.zulipchat.com/#narrow/stream/326414-t-infra.2Fbootstrap/topic/backtrace's.20size.20check.20borked.20.3A.28) would be to forcing CI to fetch the LLVM submodule at the beginning of this function.

Reply:
This feels wrong to me - the checking here runs against the rust-lang/rust checkout, right, not the submodule? So whether or not it's available we should be able to determine if things are checked out...

At the very least it seems like this is worth a comment.

(I guess it's also not clear that is_managed_git_subrepository is ever going to be *false* in CI -- that seems like a strange condition).

Reply:
> (I guess it's also not clear that is_managed_git_subrepository is ever going to be false in CI -- that seems like a strange condition).

If you mean `config.rust_info.is_managed_git_subrepository()`, it can be false for tarball sources. But for `config.in_tree_llvm_info.is_managed_git_subrepository()`, it can be false when submodules aren't fetched and `llvm.download-ci-llvm` is true.

For us, nothing changes because we always fetch submodules in our CI. This was more to help external CIs (like backtrace one — you can check their problem [here](https://rust-lang.zulipchat.com/#narrow/stream/326414-t-infra.2Fbootstrap/topic/backtrace's.20size.20check.20borked.20.3A.28)) since they might not fetch all submodules and use `download-ci-llvm=true`.

I tried to make this clear with some comments. Let me know if it needs more updates.

Reply:
Hm, I wonder if we should modify CiEnv::is_ci to *only* treat rust-lang/rust as a true CI env -- or at least provide a way to opt-out. I'm not sure that the various tweaks we do for our CI make sense to be done elsewhere, and none of them should be needed to get a build working (e.g., in backtrace's case, it sounds like they probably just want to be a regular consumer as-if they're a human on a laptop?)

Reply:
I would prefer to add an additional function to check if it's a rust-lang/rust-managed CI instead of modifying `Ci::is_ci` to keep `is_ci` logic the same for other downstream CI pipelines.